### PR TITLE
TK2.2: add service args to auth.jwt

### PR DESCRIPTION
**Pull request by frenchy64**
Epic: threatgrid/iroh#3986
Extracted from https://github.com/threatgrid/ctia/pull/924

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: TK2.2: add service args to auth.jwt
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: src/ctia/auth/jwt.clj [hunk 1]
```diff
@@ -186,19 +186,19 @@
       limit-fn)))
 
 (defn unlimited?
-  [unlimited-properties jwt]
-  (let [client-id (get jwt (iroh-claim "oauth/client/id"))
+  [unlimited-properties get-in-config jwt]
```
Comment (frenchy64) [inline src/ctia/auth/jwt.clj:189]: Added as 2nd argument because `unlimited?` is used with `partial` below.

File: test/ctia/auth/jwt_test.clj [hunk 2]
```diff
@@ -1,20 +1,24 @@
 (ns ctia.auth.jwt-test
   (:require [ctia.auth.jwt :as sut]
             [ctia.auth.capabilities :as caps]
+            [ctia.properties :as p]
             [clojure.test :as t :refer [deftest is]]
             [clojure.set :as set])
   (:import [ctia.auth.jwt JWTIdentity]))
 
+(def get-in-config
+  p/get-in-global-properties)
```
Comment (frenchy64) [inline test/ctia/auth/jwt_test.clj:10]: Stub.

File: src/ctia/auth/jwt.clj (hunk 1 above)
Comment (yogsototh) [reply src/ctia/auth/jwt.clj:189]: yep, that's also one of the reason I preferred to put the service-context as first argument and not the last.

And also to make it easier to support functions with variable number of args.
Fortunately I don't think we have much of those in the code.

**Review by yogsototh — APPROVED** [review]
LGTM!

File: src/ctia/auth/jwt.clj (hunk 1 above)
Comment (frenchy64) [reply src/ctia/auth/jwt.clj:189]: I see. This isn't a context argument though, it's a service dependency. Is this a reasonable place for it?

Comment (frenchy64) [reply src/ctia/auth/jwt.clj:189]: Merging to keep things moving, but still interested in your opinion here.

**Pull request merged**